### ISSUE TITLE
ENOENT build error with Vercel + pnpm

### ISSUE THREAD

**Issue by elliottsj** [vercel/next.js#20948]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

14.15.3

**What browser are you using?**

Edge (Chromium)

**What operating system are you using?**

macOS

**How are you deploying your application?**

Vercel

**Describe the Bug**

Context:
* https://github.com/vercel/next.js/issues/16471#issuecomment-736095201
* https://github.com/vercel/vercel/discussions/3896#discussioncomment-131265
* https://github.com/elliottsj/nextjs-vercel-pnpm-issue

As discussed above and described by @vasiliicuhar [here](https://github.com/vercel/next.js/issues/16471#issuecomment-757478987), the latest Next.js does not build on Vercel when using pnpm, due to:

```
Error: ENOENT: no such file or directory, lstat '/vercel/workpath0/app/url'
--
15:28:10.448 | at realpathSync (fs.js:1643:7)
15:28:10.448 | at handleExternals (/vercel/workpath0/app/node_modules/.pnpm/next@10.0.5_react-dom@17.0.1+react@17.0.1/node_modules/next/dist/build/webpack-config.js:66:21)
```

**Expected Behavior**

The error above should not occur, and the project should build successfully.

**To Reproduce**

A full reproduction is described here: https://github.com/elliottsj/nextjs-vercel-pnpm-issue

**Comment by vasiliicuhar** [comment]
@elliottsj how can I give it a try? Tried to use GitHub dependency, but Vercel is complaining that Next.js version is not detected

```
"dependencies": {
    "next": "elliottsj/next.js#resolve-request"
}
```

**Comment by elliottsj** [comment]
@vasiliicuhar It's a bit of an involved process: https://github.com/elliottsj/nextjs-vercel-pnpm-issue/tree/fix#how-to-repro-this-bug-fix

I thought about renaming the packages and just publishing to the canonical npm registry, but that would have been more work to make sure all the references throughout the next.js codebase are correct.

**Comment by vasiliicuhar** [comment]
I tried to drop it into a folder into my rush project, but it seems not to work well with rush 👎 (some typescript compilation errors, and complaining about react, I believe same reason why tests from this PR are failing #17882 )

I have workaround for now, but happy to give it a try when release candidate from official repo will be available.
I wonder why it works locally but doesn't on Vercel? Have you found why, or you just used your prior work?

**Comment by elliottsj** [comment]
I haven't found out exactly why `realpathSync` is having issues. Wish I could more easily debug Vercel's build runtime, but it's hard without something like SSH access.

**Comment by vasiliicuhar** [comment]
do you think it's realpath issue? On my local machine it doesn't even call handleExternals with url request.

**Comment by elliottsj** [comment]
Looks like this is fixed as of v10.0.6-canary.0 🎉 

Successful build after upgrading: https://github.com/elliottsj/nextjs-vercel-pnpm-issue/commit/1474b6fe9fc7bc05e3fe7ce8a0dabb0ae3d57289

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
